students/DrewSmith/lessons/lesson09/assignment/jpgdiscover.py

```diff
@@ -0,0 +1,29 @@
+'''
+Recursively go through directories looking for pngs
+'''
+
+
+from pathlib import Path
+
+def jpgdiscovery(directory, png_paths=None):
+    '''
+    Recursively find all PNG image files in a directory
+    :param directory: pathlib directory path to search
+    :param png_paths: list used to build return list
+    '''
+    if png_paths is None:
+        png_paths = []
+    
+    current_list = [directory, []]
+    for path in directory.iterdir():
+        if path.is_dir():
+            jpgdiscovery(path, png_paths)
+        elif path.name.endswith(".png"):
+            current_list[1].append(path.name)
+
+    if current_list[1]:
+        png_paths.extend(current_list)
+    return png_paths
```

Comment:
Very nice. I knew it would be straightforward with Paths, but I haven't played around with them enough to be able to do it off the cuff in class. Glad someone figured it out! :)